src/legacy/core_plugins/metrics/public/kbn_vis_types/request_handler.js

```diff
@@ -20,51 +20,53 @@
 import { validateInterval } from '../lib/validate_interval';
 import { timezoneProvider } from 'ui/vis/lib/timezone';
 import { timefilter } from 'ui/timefilter';
+import { kfetch } from 'ui/kfetch';
 
-const MetricsRequestHandlerProvider = function (Private, Notifier, config, $http, i18n) {
+const MetricsRequestHandlerProvider = function (Private, Notifier, config, i18n) {
   const notify = new Notifier({ location: i18n('tsvb.requestHandler.notifier.locationNameTitle', { defaultMessage: 'Metrics' }) });
+  const timezone = Private(timezoneProvider)();
 
   return {
     name: 'metrics',
-    handler: function ({ uiState, timeRange, filters, query, visParams }) {
-      const timezone = Private(timezoneProvider)();
-      return new Promise((resolve) => {
-        const panel = visParams;
-        const uiStateObj = uiState.get(panel.type, {});
-        const parsedTimeRange = timefilter.calculateBounds(timeRange);
-        const scaledDataFormat = config.get('dateFormat:scaled');
-        const dateFormat = config.get('dateFormat');
-        if (panel && panel.id) {
-          const params = {
-            timerange: { timezone, ...parsedTimeRange },
-            query,
-            filters,
-            panels: [panel],
-            state: uiStateObj
-          };
 
-          try {
-            const maxBuckets = config.get('metrics:max_buckets');
-            validateInterval(parsedTimeRange, panel, maxBuckets);
-            const httpResult = $http.post('../api/metrics/vis/data', params)
-              .then(resp => ({ dateFormat, scaledDataFormat, timezone, ...resp.data }))
-              .catch(resp => { throw resp.data; });
+    handler: ({ uiState, timeRange, filters, query, visParams }) => {
```

Comment:
converting this to an async function would allow combining the error handlers, but when looking at this I realize this doesn't do the same thing in the error situation that was being done before, is that intentional? Are kfetch errors going to render well when they happen?

Reply:
@spalger good question, I just wanted to do like before.
I think we can return rejected promise with error message  in case of exception.

Reply:
![image](https://user-images.githubusercontent.com/20072247/56592588-a8811180-65f3-11e9-9343-c53ab2599700.png)